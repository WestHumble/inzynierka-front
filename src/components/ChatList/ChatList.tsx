import React, { useContext } from "react";
import {
  View,
  Text,
  SectionList,
  StyleSheet,
  TouchableWithoutFeedback,
  Image,
} from "react-native";
import CustomButton from "../CustomButton";
import { useNavigation } from "@react-navigation/native";
import { ChatContext } from "../../context/ChatContext";
import eventButton from "../../../assets/images/right-alt-yellow.png";

const ChatItem = ({ chat }) => {
  const navigation = useNavigation();
  return (
    <View style={styles.chatContainer}>
      <View style={styles.chatDivText}>
        {chat.is_seen ? (
          ""
        ) : (
          <Text style={styles.nameNewMessage}>Nowa wiadomość</Text>
        )}
        <Text style={styles.name}>{chat.name}</Text>
      </View>
      <View style={styles.chatDivButton}>
        <TouchableWithoutFeedback
          onPress={() => navigation.navigate("Chat", { chatId: chat.id })}
        >
          <Image
            source={eventButton}
            style={styles.eventButtonImg}
            resizeMode="contain"
          />
        </TouchableWithoutFeedback>
        {/* <CustomButton
          text="Otwórz"
          onPress={() => navigation.navigate("Chat", { chatId: chat.id })}
          type="PRIMARY"
          bgColor={undefined}
          fgColor={undefined}
        /> */}
      </View>
    </View>
  );
};

const ChatList = ({ data }) => (
  <SectionList
    sections={data}
    keyExtractor={(item, index) => item.id.toString()}
    renderSectionHeader={({ section: { title } }) => (
      <View style={styles.sectionHeader}>
        <Text style={styles.sectionHeaderText}>{title}</Text>
      </View>
    )}
    renderItem={({ item }) => <ChatItem chat={item} />}
    renderSectionFooter={({ section }) =>
      section.data.length === 0 && (
        <View style={styles.noFriendsContainer}>
          <Text style={styles.noFriendsText}>Brak znajomych</Text>
        </View>
      )
    }
  />
);

const styles = StyleSheet.create({
  chatContainer: {
    padding: 16,
    marginVertical: 4,
    backgroundColor: "#131417",
    borderRadius: 20,
    shadowColor: "black",
    shadowOffset: { width: 0, height: -3 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    flexDirection: "row",
    alignItems: "center",
  },
  name: {
    fontSize: 18,
    fontWeight: "bold",
    color: "#fff",
  },
  sectionHeader: {
    color: "#fff",
    padding: 8,
    marginTop: 12,
  },
  sectionHeaderText: {
    fontSize: 21,
    fontWeight: "bold",
    color: "#fff",
  },
  noFriendsContainer: {
    padding: 16,
    alignItems: "center",
  },
  noFriendsText: {
    fontSize: 16,
    color: "#888",
  },
  chatDivText: {
    flex: 2 / 3,
    flexDirection: "column",
    justifyContent: "space-between",
  },
  chatDivButton: {
    flex: 1 / 3,
    flexDirection: "row",
    justifyContent: "flex-end",
  },
  eventButtonImg: {
    width: 30,
    height: 30,
  },
  nameNewMessage: {
    color: "#ffffff",
    backgroundColor: "#F2B138",
    padding: 5,
    borderRadius: 10,
    marginBottom: 5,
    width: 125,
  },
});

export default ChatList;
