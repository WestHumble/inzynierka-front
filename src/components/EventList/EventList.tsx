import React from "react";
import {View, Text, Image, SectionList, StyleSheet, TouchableWithoutFeedback} from "react-native";
import eventButton from "../../../assets/images/right-alt-yellow.png";
import {useNavigation} from "@react-navigation/native";

const EventItem = ({ date, month, title, hour, id }) => {
  const navigation = useNavigation()
  const onEventPressed = () => {
    navigation.navigate("EventView", {eventId: id})
  }
  return (
      <View style={styles.eventContainer}>
        <View style={styles.eventDate}>
          <Text style={styles.month}>{month}</Text>
          <Text style={styles.date}>{date}</Text>
        </View>
        <View style={styles.eventContent}>
          <Text style={styles.hour}>{hour}</Text>
          <Text style={styles.title}>{title}</Text>
        </View>
        <View style={styles.eventButton}>
          <TouchableWithoutFeedback  onPress={onEventPressed}>
            <Image
                source={eventButton}
                style={styles.eventButtonImg}
                resizeMode="contain"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
  );
}

function konwerterDaty(data) {
  const months = [
    "Sty",
    "Lut",
    "Mar",
    "Kwi",
    "Maj",
    "Cze",
    "Lip",
    "Sie",
    "Wrz",
    "Paź",
    "Lis",
    "Gru",
  ];
  const month = months[data.getMonth()];
  return month;
}

const EventList = ({ data }) => (
  <SectionList
    sections={data}
    stickySectionHeadersEnabled={false}
    keyExtractor={(item, index) => index.toString() + item.id.toString()}
    renderSectionHeader={({ section: { title } }) => (
      <View style={styles.sectionHeader}>
        <Text style={styles.sectionHeaderText}>{title}</Text>
      </View>
    )}
    renderItem={({ item }) => (
      <EventItem
        id = {item.id}
        month={konwerterDaty(new Date(item.date.date))}
        date={new Date(item.date.date).toLocaleString([], {
          day: "numeric",
        })}
        title={item.name}
        hour={new Date(item.date.date).toLocaleString([], {
          hour: "2-digit",
          minute: "2-digit",
        })}
      />
    )}
    renderSectionFooter={({ section }) =>
      section.data.length === 0 && (
        <View style={styles.noEventsContainer}>
          <Text style={styles.noEventsText}>Brak wydarzeń</Text>
        </View>
      )
    }
  />
);

const styles = StyleSheet.create({
  eventContainer: {
    padding: 16,
    marginVertical: 4,
    backgroundColor: "#131417",
    borderRadius: 20,
    shadowColor: "black",
    shadowOffset: { width: 0, height: -3 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    flexDirection: "row",
  },
  title: {
    fontSize: 18,
    fontWeight: "bold",
    color: "#fff",
  },
  date: {
    color: "#fff",
    fontSize: 24,
    fontWeight: "bold",
  },
  month: {
    color: "#fff",
    fontSize: 18,
    fontWeight: "bold",
  },
  hour: {
    color: "#F2B138",
  },
  location: {
    color: "#888",
  },
  description: {
    marginTop: 8,
  },
  sectionHeader: {
    color: "#fff",
    padding: 8,
    marginTop: 12,
  },
  sectionHeaderText: {
    fontSize: 21,
    fontWeight: "bold",
    color: "#fff",
    paddingVertical: 8,
    marginVertical: 4,
  },
  noEventsContainer: {
    padding: 16,
    alignItems: "center",
  },
  noEventsText: {
    fontSize: 16,
    color: "#888",
  },
  eventDate: {
    marginRight: 16,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#F2B138",
    padding: 8,
    width: 70,
    height: 70,
    borderRadius: 10,
  },
  eventContent: {
    flex: 1,
    alignContent: "center",
    justifyContent: "center",
  },
  eventButton: {
    justifyContent: "center",
    alignItems: "center",
  },
  eventButtonImg: {
    width: 30,
    height: 30,
  },
});

export default EventList;
