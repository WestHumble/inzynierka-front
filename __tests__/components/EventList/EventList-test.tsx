import React, {useState} from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from '@jest/globals';
import EventList from "../../../src/components/EventList";
import ChatList from "../../../src/components/ChatList";

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});
test('empty', () => {
  const tree = renderer.create(
      <ChatList data={[
        {
          title: "Chaty",
          data: [],
        },
      ]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders empty', () => {
  const tree = renderer.create(<EventList data={[
    {
      title: "Moje",
      data: [],
    },
    {
      title: "Zaproszone",
      data: [],
    },
    {
      title: "Publiczne",
      data: [],
    },
  ]} />).toJSON();
  expect(tree).toMatchSnapshot();
});


test('renders with items', () => {
  const tree = renderer.create(<EventList data={[
    {
      title: "Moje",
      data: [{
        id: 1,
        user_id: 1,
        latlng: null,
        name: "name",
        description: "description",
        title: "title",
        date: "2024-01-01 11:22",
        address: "address",
        invite_status: "new",
      }],
    },
    {
      title: "Zaproszone",
      data: [{
        id: 2,
        user_id: 1,
        latlng: null,
        name: "name",
        description: "description",
        title: "title",
        date: "2024-01-01 11:22",
        address: "address",
        invite_status: "new",
      }],
    },
    {
      title: "Publiczne",
      data: [{
        id: 3,
        user_id: 1,
        latlng: null,
        name: "name",
        description: "description",
        title: "title",
        date: "2024-01-01 11:22",
        address: "address",
        invite_status: "new",
      }],
    },
  ]} />).toJSON();
  expect(tree).toMatchSnapshot();
});