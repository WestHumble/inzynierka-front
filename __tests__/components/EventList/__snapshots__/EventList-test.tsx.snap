// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [],
        "title": "Chaty",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Chaty
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 16,
            }
          }
        >
          Brak wydarzeń
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders empty 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [],
        "title": "Moje",
      },
      {
        "data": [],
        "title": "Zaproszone",
      },
      {
        "data": [],
        "title": "Publiczne",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Moje
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 16,
            }
          }
        >
          Brak wydarzeń
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Zaproszone
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 16,
            }
          }
        >
          Brak wydarzeń
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Publiczne
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 16,
            }
          }
        >
          Brak wydarzeń
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders with items 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [
          {
            "address": "address",
            "date": "2024-01-01 11:22",
            "description": "description",
            "id": 1,
            "invite_status": "new",
            "latlng": null,
            "name": "name",
            "title": "title",
            "user_id": 1,
          },
        ],
        "title": "Moje",
      },
      {
        "data": [
          {
            "address": "address",
            "date": "2024-01-01 11:22",
            "description": "description",
            "id": 2,
            "invite_status": "new",
            "latlng": null,
            "name": "name",
            "title": "title",
            "user_id": 1,
          },
        ],
        "title": "Zaproszone",
      },
      {
        "data": [
          {
            "address": "address",
            "date": "2024-01-01 11:22",
            "description": "description",
            "id": 3,
            "invite_status": "new",
            "latlng": null,
            "name": "name",
            "title": "title",
            "user_id": 1,
          },
        ],
        "title": "Publiczne",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Moje
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "flexDirection": "row",
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F2B138",
              "borderRadius": 10,
              "height": 70,
              "justifyContent": "center",
              "marginRight": 16,
              "padding": 8,
              "width": 70,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          />
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Invalid Date
          </Text>
        </View>
        <View
          style={
            {
              "alignContent": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#F2B138",
              }
            }
          >
            Invalid Date
          </Text>
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            name
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Image
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="contain"
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Zaproszone
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "flexDirection": "row",
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F2B138",
              "borderRadius": 10,
              "height": 70,
              "justifyContent": "center",
              "marginRight": 16,
              "padding": 8,
              "width": 70,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          />
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Invalid Date
          </Text>
        </View>
        <View
          style={
            {
              "alignContent": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#F2B138",
              }
            }
          >
            Invalid Date
          </Text>
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            name
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Image
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="contain"
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
              "marginVertical": 4,
              "paddingVertical": 8,
            }
          }
        >
          Publiczne
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "flexDirection": "row",
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F2B138",
              "borderRadius": 10,
              "height": 70,
              "justifyContent": "center",
              "marginRight": 16,
              "padding": 8,
              "width": 70,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          />
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Invalid Date
          </Text>
        </View>
        <View
          style={
            {
              "alignContent": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#F2B138",
              }
            }
          >
            Invalid Date
          </Text>
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            name
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Image
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="contain"
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
